includes/class-wc-mondido-hpos-compatibility.php

```diff
@@ -0,0 +1,68 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+} // Exit if accessed directly
+
+use Automattic\WooCommerce\Utilities\OrderUtil;
+
+interface OrderStorageInterface {
```

Comment:
Having `Interface` in the name is ugly and violates the IS-A relationship to implementing classes. For instance, `HPOSOrderStorage` implements `OrderStorageInterface` but it is wrong to say that it "is a" `OrderStorageInterface`. `HPOSOrderStorage` is not an interface it is a kind of order storage. The same is true for `CPTOrderStorage`. Since both those classes are examples of order storage, it makes sense to use the name `OrderStorage` for the shared interface.

If the problem is that we already use the name `OrderStorage` as a placeholder for the `current()` method, my suggestion is that we merge `OrderStorageInterface` and `OrderStorage` into an abstract class named `OrderStorage`.
```
abstract class OrderStorage
{
	public static function current(): OrderStorage
	{
		if (OrderUtil::custom_orders_table_usage_is_enabled()) {
			return new HPOSOrderStorage();
		} else {
			return new CPTOrderStorage();
		}
	}

	public abstract function get_meta_data($order, $meta_key, $meta_value);
	public abstract function add_meta_data($order, $meta_key, $meta_value);
	public abstract function update_meta_data($order, $meta_key, $meta_value);
	public abstract function delete_meta_data($order, $meta_key);
	public abstract function save($order);
}
```

Reply:
That's exactly the issue I saw with clashing names. Thanks for the input, learning a lot about OOP best practices here, great stuff :)